app/assets/javascripts/discourse/components/site-header.js.es6

```diff
@@ -23,6 +28,135 @@ const SiteHeaderComponent = MountWidget.extend(Docking, {
     this.queueRerender();
   },
 
+  _panOpenClose(offset, velocity, direction) {
+    const $window = $(window);
+    const windowWidth = parseInt($window.width());
+    const $menuPanels = $(".menu-panel");
+    direction === "close" ? (offset += velocity) : (offset -= velocity);
+    const menuOrigin = this.get("panMenuOrigin");
+    $menuPanels.each((idx, panel) => {
+      const $panel = $(panel);
+      const $headerCloak = $(".header-cloak");
+      $panel.css(menuOrigin, -offset);
+      $headerCloak.css("opacity", Math.min(0.5, (300 - offset) / 600));
+      if (offset > windowWidth) {
+        $panel.css(menuOrigin, -windowWidth);
+        this.set("animate", true);
+        this.eventDispatched("dom:clean", "header");
+        this.set("panMenuOffset", 0);
+      } else if (offset <= 0) {
+        $panel.css("right", "");
+        $panel.css("left", "");
+        this.set("panMenuOffset", 0);
+      } else {
+        Ember.run.later(
```

Comment:
I wonder if this event should check if one is already scheduled and clear it first? 20ms is pretty fast but is there any chance two could be enqueued at once? I would recommend storing the timer in a private variable and calling Ember.run.cancel before.

You should also clear the timeout in `willDestroyElement`